zinc/src/main/scala/sbt/internal/inc/MixedAnalyzingCompiler.scala

```diff
@@ -181,13 +195,28 @@ object MixedAnalyzingCompiler {
       incrementalCompilerOptions: IncOptions,
       extra: List[(String, String)]
   ): CompileConfiguration = {
-    val classpathHash = classpath map { x =>
-      FileHash.of(x, Stamper.forHash(x).hashCode)
+    // #433: Cache jars with their metadata to avoid recomputing hashes transitively in other projects
+    val parallelClasspathHashing = classpath.toParArray.map { file =>
+      if (!file.exists()) emptyFileHash(file)
```

Comment:
I'm pretty sure you can use the file attributes on a non-existent file and then use the response to say... if that is the case it would be best to use it here and avoid a stack frame (which is slow on windows so best to batch / minimise all I/O calls)

Reply:
I'm pretty sure you can't, I was just getting IO errors because of this.

Reply:
@fommil I got a proof, `/caca/peluda` does not exist:
```scala
scala> val attrs = java.nio.file.Files.readAttributes((new java.io.File("/caca/peluda").toPath), classOf[BasicFileAttributes])
java.nio.file.NoSuchFileException: /caca/peluda
  at sun.nio.fs.UnixException.translateToIOException(UnixException.java:86)
  at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:102)
  at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:107)
  at sun.nio.fs.UnixFileAttributeViews$Basic.readAttributes(UnixFileAttributeViews.java:55)
  at sun.nio.fs.UnixFileSystemProvider.readAttributes(UnixFileSystemProvider.java:144)
  at sun.nio.fs.LinuxFileSystemProvider.readAttributes(LinuxFileSystemProvider.java:99)
  at java.nio.file.Files.readAttributes(Files.java:1737)
  ... 28 elided
```

Reply:
ok cool, that's good to know.